### PR TITLE
Add SMTP Relay support

### PR DESCRIPTION
**Pull request by cmstew**
Adds smarthost capabilities to Hestia.
- You may optionally set a global smarthost
- You may also optionally set a different smarthost per domain.

### CODE REVIEW

File: install/upgrade/versions/1.3.3.sh [hunk 1]
```diff
@@ -16,4 +16,92 @@ if [ -e "/etc/nginx/nginx.conf" ]; then
...
+        EOL
+        )
+
+        if [ "$disable_smarthost" = true ]; then
+            insert=$(sed 's/^/#/g; s/\\n/\\n#/g; s/.$//' <<< $insert)
+        fi
+
+        line=$(expr $(sed -n '/begin transports/=' /etc/exim4/exim4.conf.template) + 2)
+        sed -i "${line}i $insert" /etc/exim4/exim4.conf.template
+    fi
```
Comment (jaapmarcus) [inline install/upgrade/versions/1.3.3.sh]: Whole script didn't get called during the upgrade we should rename everything to 1.4.0.sh

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

**Comment by jaapmarcus** [comment]
For install/upgrade/versions/1.3.3.sh

This has been discussed in Discord we should  rename it to 1.4.0.sh but the upgrade script didn't call.

**Comment by jaapmarcus** [comment]
Entering the password is still required when updating webmail system / dkim / ssl
Screenshot 2021-01-30 at 09 16 28

**Comment by jaapmarcus** [comment]
I haven't checked the code yet:

Screenshot 2021-01-30 at 09 18 37


This is made before changing

**Comment by jaapmarcus** [comment]
https://github.com/cmstew/hestiacp/blob/d30ed2c4b7e16f22178e69e99a661957850fc687/bin/v-add-mail-domain-smtp-relay#L56-L65

Add check if vars all ready exist currently they are added multiple times

```
 U_SMARTHOST='yes' U_SMARTHOST_HOST='mail.smtp2go.com' U_SMARTHOST_PORT='587' U_SMARTHOST_USERNAME='xxx' U_SMTP_RELAY='true' U_SMTP_RELAY_HOST='mail.smtp2go.com' U_SMTP_RELAY_PORT='2525' U_SMTP_RELAY_USERNAME='xxx'
```

We should also save the password incase of a restore....

**Comment by cmstew** [comment]
> Entering the password is still required when updating webmail system / dkim / ssl
> Screenshot 2021-01-30 at 09 16 28

This is actually resolved now.

**Comment by cmstew** [comment]
> https://github.com/cmstew/hestiacp/blob/d30ed2c4b7e16f22178e69e99a661957850fc687/bin/v-add-mail-domain-smtp-relay#L56-L65
> 
> Add check if vars all ready exist currently they are added multiple times
> 
> ```
>  U_SMARTHOST='yes' U_SMARTHOST_HOST='mail.smtp2go.com' U_SMARTHOST_PORT='587' U_SMARTHOST_USERNAME='xxx' U_SMTP_RELAY='true' U_SMTP_RELAY_HOST='mail.smtp2go.com' U_SMTP_RELAY_PORT='2525' U_SMTP_RELAY_USERNAME='xxx'
> ```
> 
> We should also save the password incase of a restore....

- Added checks for vars
- Added saving password
- Added restore capability

**Review by jaapmarcus — APPROVED** [review]

**Comment by jaapmarcus** [comment]
Changes are working fine.

Pushed 2 small changes when password of smtp relay was changed it didn't update propperly

**Pull request merged**